libnmstate/validator.py

```diff
@@ -334,3 +341,61 @@ def _assert_ovs_lag_slave_count(iface_state):
                 raise NmstateOvsLagValueError(
                     f"OVS {ifname} LAG port {lag} has less than 2 slaves."
                 )
+
+
+def _validate_linux_bond(original_desired_state, current_state):
+    """
+    Raise NmstateValueError on these scenarios:
+        * Bond mode not defined.
+        * Original desire state contains illegal bond config.
+        * After merge, user's intention will cause illegal bong config.
```

Comment:
Just mention it here again, since the commend would need to be updated:

IMHO it should also be illegal if the current_state has a hardcoded MAC and the user specifies fail_over_mac/active-backup. But this is up for discussion.